### ISSUE TITLE
Bug with uppercase `# noqa: SIM114` and RUF100

### ISSUE THREAD

**Issue by sondrfos** [astral-sh/ruff#6025]
Hi,

Thanks for making Ruff, it's an exellent linter :) 

We just recently updated our project to Ruff 0.0.280 and seems to experience a weird bug.
Our code uses a rather large if-statement pulled straight form a research paper. To keep the code and the paper pretty equal, the code is written the exact same way as the paper. This means that we need to ignore some SIM114-warnings. 

When updating to 0.0.280 we started to receive a warning
 ```RUF100 [*] Unused `noqa` directive (unused: `SIM114`)``` 
However, when removing the `# noqa: SIM114` we start to receive the warning:
```SIM114 Combine `if` branches using logical `or` operator```

When testing we discovered that if we changed to the comment to `# noqa: sim114` we dont receive the RUF100 anymore.

I'll include some small example code which shows the weird behaviour:

```python
a = True
b = False

def foo():
    if a > b: # noqa: SIM114
        return 3
    elif a == b:
        return 3
    elif a <  b: # noqa: sim114
        return 4
    elif b is None:
        return 4
```
The first `noqa` gives a RUF100 warning, the second gives none.

**Comment by charliermarsh** [comment]
Will take a look today!

**Comment by charliermarsh** [comment]
Okay, it's a few confusing things working together... The root of the problem is that if branches contain different comments, we avoid flagging `SIM114`. So when you have this:

```python
def foo():
    if a > b: # noqa: SIM114
        return 3
    elif a == b:
        return 3
```

We _don't_ flag `SIM114` _because_ of the `# noqa: SIM114` comment, and so the comment then becomes unused. If you remove the comment, then `SIM114` is raised again, because the comment is no longer present to _make_ the branches non-identical.

**Comment by charliermarsh** [comment]
I'm going to change the logic to ignore end-of-line comments and only check in-body comments, which will remove this confusion. (Your code with `# noqa: SIM114` at the end of the test lines should work as expected.)

**Comment by sondrfos** [comment]
That makes sense. thanks for such quick responses! : )
